Admin/AbstractAdmin.php

```diff
@@ -3112,7 +3112,11 @@ public function canAccessObject($action, $object)
         );
 
         if ($this->adminBuilder === null) {
-            $this->adminBuilder = new AdminBuilder();
+            $this->adminBuilder = new AdminBuilder(
+                $this->getShowBuilder(),
+                $this->getListBuilder(),
+                $this->getDatagridBuilder()
```

Comment:
AFAIK there is the binding for this classes:
https://github.com/sonata-project/SonataAdminBundle/blob/3.x/DependencyInjection/Compiler/AddDependencyCallsCompilerPass.php#L258-L260